assets/src/components/ladderPage.tsx

```diff
@@ -154,6 +156,16 @@ const LadderPage = (): ReactElement<HTMLDivElement> => {
   const timepointsByRouteId: TimepointsByRouteId =
     useTimepoints(selectedRouteIds)
 
+  const { socket } = useSocket()
+  const alerts = useAlerts(socket, selectedRouteIds)
+  const routesWithAlerts = []
+
+  for (const routeId in alerts) {
+    if (alerts[routeId].length > 0) {
+      routesWithAlerts.push(routeId)
+    }
+  }
+
```

Comment:
question: why map/collect alerts here instead of passing `alerts[routeId]` inside of `routeLadders`?

To me it seems that having the actual alert objects within the route ladder could be useful in the future if we want to provide more context then the `"Active detour"` tooltip. We could also provide the number of alerts on a route if we passed the alerts array to the route ladder.

Additionally, if the routes with alerts default to an empty array, and the `.length > 0` check is moved into the route ladder component, then the property doesn't need to be set for all the tests that didn't have it before?

But maybe there's something else or something I'm missing?

Reply:
I was taking a sort of conservative "pass only the data that the component really needs" approach, but you're definitely right that we might want to access some of that information in the future, so I'm open to the idea of changing it.

> if the routes with alerts default to an empty array, and the .length > 0 check is moved into the route ladder component, then the property doesn't need to be set for all the tests that didn't have it before?

I think I'm missing something here, could you elaborate?

Reply:
> I was taking a sort of conservative "pass only the data that the component really needs" approach
Okay cool that's what I figured 👍🏻. 

> > if the routes with alerts default to an empty array, and the .length > 0 check is moved into the route ladder component, then the property doesn't need to be set for all the tests that didn't have it before?
> 
> I think I'm missing something here, could you elaborate?

Yeah it's strangely phrased and I don't think it was necsisary anyway.
I was trying to communicate that the codeblock this comment is attached to (and _maybe_ some of the supporting infrastructure) could disapear if the `hasAlert` property could be defined as `bool | null` and the route ladder could treat the null/undefined case also as `false`, which would reduce the extra changes needed to add `hasAlert` to all relavent tests, but that comes down to preference and shouldn't hold up merging.